src/core/utils/api-client/notifications.ts

```diff
@@ -22,25 +23,37 @@ export class NotificationsApiClient {
      * TODO: When fetching multiple pages of notifications, maybe we should find a way to cache those notifications
      *       in order to minimise the calls to our api
      */
-    public async fetchNotifications(walletPublicKey: string, page?: number) {
+    public async fetchNotifications(
+        walletPublicKeys: string[],
+        blockchainNetworks: { blockchain: Blockchain; chainId: string }[],
+        page?: number
+    ) {
         try {
-            const walletPrivateKey = await getWalletCredentialsKey(walletPublicKey);
-            if (walletPrivateKey) {
-                const data: any = {
+            const data = {
+                walletPublicKeys: [],
+                blockchainNetworks,
+                page: page || 1
+            };
+
+            for (const walletPublicKey of walletPublicKeys) {
+                const walletPrivateKey = await getWalletCredentialsKey(walletPublicKey);
+
+                const walletData: any = {
                     walletPublicKey,
                     timestamp: await getCurrentTimestampNTP(),
```

Comment:
does this have a cache ? or it does a ntp call every time ?

Reply:
It does make a ntp call every time.
Any suggestions on how to cache?

Reply:
yep, quite simple: `const timestamp = await getCurrentTimestampNTP()` before `for` :) this way you do it once :P